qiskit/qobj/converters/pulse_instruction.py

```diff
@@ -544,21 +550,27 @@ def convert_set_frequency(self, instruction):
 
         Args:
             instruction (PulseQobjInstruction): set frequency qobj instruction
+                                                The passed frequency will be scaled by a factor 1e9
         Returns:
             Schedule: Converted and scheduled Instruction
         """
         t0 = instruction.t0
         channel = self.get_channel(instruction.ch)
-        frequency = instruction.frequency * 1e9
+        frequency = instruction.frequency
 
         if isinstance(frequency, str):
             frequency_expr = parse_string_expr(frequency, partial_binding=False)
 
             def gen_sf_schedule(*args, **kwargs):
                 _frequency = frequency_expr(*args, **kwargs)
-                return instructions.SetFrequency(_frequency, channel) << t0
+                return instructions.SetFrequency(_frequency * FREQUENCY_SCALING_FACTOR,
+                                                 channel) << t0
 
-            return ParameterizedSchedule(gen_sf_schedule, parameters=frequency_expr.params)
+            schedule = ParameterizedSchedule(parameters=frequency_expr.params)
+            schedule.add_parameterized(gen_sf_schedule)
+            return schedule
+        else:
+            frequency = frequency * FREQUENCY_SCALING_FACTOR
```

Comment:
@taalexander I'm a bit worried here about this happening under the hood... I added something in the docs and a constant, to try and make this clearer and more consistent, but let me know if you'd like a different solution.

Reply:
This is a result of Qiskit using Si units and the backend using GHz, the format of each is defined in the schema/the Qiskit objects.